test/sample.zsh_history

```diff
@@ -0,0 +1,33 @@
+: 1492369835:0;hist import
+: 1492369844:0;hist /hist import
+: 1492369861:4;hist shell
+: 1492369868:0;hist version
+: 1492369908:0;hist /hist | wc -l
+: 1492370096:30;history
+: 1492370103:0;man history
+: 1492370245:0;history-stat
+: 1492370604:50;vi $HISTFILE
+: 1492370674:0;tail $HISTFILEfoo
+: 1496750990:0;{ printf ' }\
```

Comment:
I actually first tried to go for 100% accurate command parsing with `bash -n` (which parses syntax and only returns 0 if it's ok and something else otherwise) but when I tested some potential counter-examples, I found this. This does not parse according to `bash -n` but bash nevertheless tries to execute it and stores it in the history file.